eralchemy2/models.py

```diff
@@ -1,56 +1,62 @@
+"""
+All the intermediary syntax.
+We can several kinds of models can be translated to this syntax.
+"""
+
+from __future__ import annotations
+
 import operator
 import re
+from abc import ABC, abstractmethod
+from typing import Any, ClassVar, TypeVar
 
 from .cst import FONT_TAGS, ROW_TAGS, TABLE
 
-"""
-All the intermediary syntax.
-We can several kinds of models can be translated to this syntax.
-"""
+T = TypeVar("T")
 
 
-class Drawable:
+class Drawable(ABC):
     """Abstract class to represent all the objects which are drawable."""
 
-    RE = None
+    RE: ClassVar[re.Pattern[str]]
 
-    def to_markdown(self):
+    def to_markdown(self) -> str:
         """Transforms the intermediary object to it's syntax in the er markup."""
-        raise NotImplemented()
+        raise NotImplementedError()
 
-    def to_dot(self):
+    def to_dot(self) -> str:
         """Transforms the intermediary object to it's syntax in the dot format."""
-        raise NotImplemented()
+        raise NotImplementedError()
 
-    def __eq__(self, other):
+    def __eq__(self, other) -> bool:
         return self.__dict__ == other.__dict__
 
-    @staticmethod
-    def make_from_match(self):
+    @classmethod
+    @abstractmethod
+    def make_from_match(cls: type[T], match: re.Match) -> T:
```

Comment:
same here - shouldn't this stay a staticmethod?

Reply:
To allow typing (see type[T] and T)

Reply:
but can be removed

Reply:
But what is the benefit of `type[T]` here?
Doing it with `-> Drawable` as it is now is what I would have thought?

Looks good!